src/error.rs

```diff
@@ -722,7 +721,7 @@ impl error::Error for CertError {}
 /// Errors which may be encountered when handling attestation reports
 pub enum AttestationReportError {
     /// Bincode Error Handling
-    BincodeError(bincode::ErrorKind),
+    BincodeError(bincode::error::DecodeError),
```

Comment:
Why not map this to the `BincodeError` you created? It could map to `BincodeError::DecodeError`.